rfcs/text/0000-tech-planning-process.md

```diff
@@ -0,0 +1,81 @@
+
+## Definitions
+
+**Milestones** represent a four week timeline/duration in a Release.
+
+**Releases** are composed of three milestones and occur on a quarterly cadence.
+
+**Technical Objective:** a high level goal that encapsulates a general area of DCP technical work that can be flexibly broken up by specific sub-tasks. Sub-tasks are usually represented as Epics. More details on the definition is below.
+
+**Technical Objectives RFC:** The final artifact that will be consulted by the PM team for the roadmap refresh, with the expectation that the objectives will be prioritized for the following quarter, following spirited discussion as needed. 
+
+## Detailed Design
+
+### What is a Technical Objective?
+
+A Technical Objective is a body of work for the DCP, usually cross-cutting, related to:
+
+1) Scalability and Performance
+2) Reduction of code debt to improve reliability
+3) Deprecation (or migration) of technologies used in the DCP
+4) Refactoring work in order to make modifications to the DCP Architecture (usually related to points 1 or 2).
+
+A Technical Objective is NOT:
+
+1) A “feature” related to the DCP product. If a TL deems that some feature is missing from the Product Roadmap, it is the responsibility of the TL to communicate with someone on the PM team to ensure that the feature's need is brought up in PM discussions.
+2) Related to DevSecOps or Compliance. Objectives that fall under either of these categories include anything related to CI/CD, security, and Legal (e.g. GDPR). The DevSecOps team and the Compliance working groups are responsible for establishing these objectives for the Product Roadmap.
+
+### Process
+
+Per the [Roadmaps and Planning Process Timeline](https://github.com/HumanCellAtlas/dcp-community/blob/master/rfcs/text/0012-roadmaps%2Bplanning.md#timeline-for-roadmaps), a the Technical Objectives RFC is expected to be completed by the end of Milestone 2, Week 1 (M2W1) which informs the timeline detailed below.
+
+| Week in Milestone | Deliverables |
+|           :----:                    | :--- |
+| **M1W1**  | A Google Doc is sent out (either the previous document used from a previous quarter or a fresh document) to the Technical Architecture team in order to solicit input on what are deemed to be important technical areas of improvement that should be prioritized in the upcoming quarter. The input to this document is free-form and TLs may or may not choose to participate, with the understanding that if no participation is given, no objectives will be generated. |
+| **M1W2** | The Technical Architecture Chairperson (Chair) will spend the week coalescing the free-form text into concrete objectives that will be presented in the Technical Architecture meeting. A discussion will ensue to form consensus around the objectives and if no consensus is met, the entire set of objectives will be included in the RFC for further debate during the RFC's community review. |
```

Comment:
I recommend the consensus part is spread over 2 weeks rather than one, so that someone who happens to absent one week still has a chance to comment on the second. Or this could be left to the RFC period. What do you think? Otherwise lgtm.

Reply:
Hmm maybe my language isn't clear here? That's actually what I was going for.. M1W1 was part one of consensus and M1W2 was week 2 of consensus building? I was trying to say that if somehow it's done by W1, then we can skip week 2.